### PR TITLE
PR:  Some indentation fixes

### LINKED ISSUE #4475: Inserting a line-break in code/comment with parenthesis deletes part of the code/comment
## Description of your problem

I have a comment that is too long so I want to break it into two lines - this comment also contains parenthesis which seems to be causing the bug.  I go to where I want to break it and when I hit enter, it chops off some of the text from the line that is moved down a row.  Removing the parenthesis stops the strange behavior.  The example should explain it.

**What steps will reproduce the problem?**

1. enter this comment with all the preceding tabs (it's wrapped when I entered it):

```
								#verify that file has station name in it (just to be sure we're not accidentally pulling the wrong station)
```

2. place your cursor just in front of "pulling" and hit enter

3. you will end up with this:

```
								#verify that file has station name in it (just to be sure we're not accidentally 
												  lling the wrong station)
```

4. As you can see "pulling" has been chopped and it also has the incorrect indentation.  Now put your cursor in front of "lling" and hit backspace a lot of times to try and get "lling" to merge back up with the previous line - it will get "stuck" at the left margin.  It will look like this:

```
								#verify that file has station name in it (just to be sure we're not accidentally 
  lling the wrong station)
```

**What is the expected output? What do you see instead?**

It should insert a clean line break, properly tabbed.  Backspace should be able to merge it with the previous line.

**Please provide any additional information below**

## Versions and main components

* Spyder Version: v3.1.4
* Python Version: 3.6.0 64bit
* Qt Version: 5.6.2
* PyQt Version: 5.6
* Operating system: Windows 10

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:

jedi >=0.9.0     :  0.9.0 (OK)
matplotlib >=1.0 :  2.0.0 (OK)
nbconvert >=4.0  :  4.2.0 (OK)
numpy >=1.7      :  1.11.3 (OK)
pandas >=0.13.1  :  0.19.2 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.1.3 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.5.1 (OK)
sympy >=0.7.3    :  1.0 (OK)

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes #4475

- indentation chopping text when indent with tabs
- inline comments breaking indentation

no xfail tests in auto_indentation 😄🙈

### CODE REVIEW

File: spyder/widgets/sourcecode/codeeditor.py [hunk 1]
```diff
@@ -1925,6 +1925,11 @@ def fix_indent(self, forward=True, comment_or_string=False):
             cursor.movePosition(QTextCursor.PreviousBlock)
             prevtext = to_text_string(cursor.block().text()).rstrip()
 
+            # Remove inline comment
+            inline_comment = prevtext.find('#')
```
Comment (ccordoba12) [inline spyder/widgets/sourcecode/codeeditor.py]: Is this enough to detect an inline comment? What if you have a line like

    # foo

?

Comment (rlaverde) [inline spyder/widgets/sourcecode/codeeditor.py]: `prevtext = ""` and the indentation will be calculated to `0` as expected.

I was testing and It fails in expressions with `#` that isn't a comment, and the crop cause to drop a bracket, although it is pretty rare (e. g. a list assignment to a dict element that contains a "#"):

`a_dict["#"] = (`

```
a_dict["#"] = (
       |
```

**Comment by ccordoba12** [comment]
@rlaverde, this fixes all issues mentioned in #4475?

**Comment by rlaverde** [comment]
@rlaverde, this fixes all issues mentioned in #4475?
> yes, they are related with auto indentation chopping text when indenting with tabs.

**Comment by ccordoba12** [comment]
Ok, let's merge it then.

**Pull request merged**